components/raftstore-v2/src/operation/query/capture.rs

```diff
@@ -52,14 +52,16 @@ impl<'a, EK: KvEngine, ER: RaftEngine, T: raftstore::store::Transport>
     pub fn on_capture_change(&mut self, capture_change: CaptureChange) {
         fail_point!("raft_on_capture_change");
 
-        let apply_router = self.fsm.peer().apply_scheduler().unwrap().clone();
+        let apply_scheduler = self.fsm.peer().apply_scheduler().cloned();
         let (ch, _) = QueryResChannel::with_callback(Box::new(move |res| {
             if let QueryResult::Response(resp) = res && resp.get_header().has_error() {
                 // Return error
                 capture_change.snap_cb.report_error(resp.clone());
                 return;
             }
-            apply_router.send(ApplyTask::CaptureApply(capture_change))
+            if let Some(scheduler) = apply_scheduler {
```

Comment:
When apply_scheduler is None, ApplyTask::CaptureApply will not be sent. Will it be a problem? @tabokie @overvenus

Reply:
Yes, CDC may hang.

I wonder when it will be `None`?

Reply:
When peer is being destroyed.

Reply:
@overvenus Then will TiCDC be able to recover from it?

Reply:
Maybe, it should get a `RegionChangeEvent::Destroy` event.
Anyway, it's better to invoke callback actively.

```rust
if let Some(scheduler) = apply_scheduler { ... } else {
    capture_change.snap_cb.report_error(region_not_found)
}
```